apache/superset
# fix(sqllab): remove link to sqllab if missing perms

### SUMMARY
Currently the "View in SQL Lab" option is available in the Dataset menu in Explore view even when the user doesn't belong to the `sql_lab` role. In addition, SQL Lab is accessible (albeit doesn't work properly) when the necessary permissions are missing.

This PR does the following:
- Removes the "View in SQL Lab" option from the Dataset menu
- Redirects the user to the welcome page when trying to access SQL Lab without necessary permissions.
- Remove "Saved Queries" from Welcome page if user is missing `sql_lab` role

### AFTER
Now when a user that doesn't have the `sql_lab` role will no longer see the "View in SQL Lab" menu item:

![image](https://user-images.githubusercontent.com/33317356/210236216-67179de5-1e3b-45d7-9d07-9a34e512f5b6.png)

For admin and users with `sql_lab` role the menu is unchanged:

![image](https://user-images.githubusercontent.com/33317356/210236242-91967cc4-371b-4277-9266-28848034bd8f.png)

When attempting to access SQL Lab without correct perms via a direct URL, the user gets redirected to the welcome page with an error toast. Also, the welcome page doesn't show the "Saved Queries" section if the user doesn't belong to the `sql_lab` role (we could optionally check the specific perms for saved queries, but maybe we can do that in a follow up if that's really needed):

![image](https://user-images.githubusercontent.com/33317356/211002955-444661a6-e55f-4c2c-9a93-96bbb0de40ce.png)

Note that due to the way we're currently caching bootstrap data, toasts don't always appear correctly (they can appear delayed or not get cleared after displaying). I will follow up separately to fix this.

### TESTING INSTRUCTIONS
For testing instructions, please see the following comment in this PR: https://github.com/apache/superset/pull/22566#issuecomment-1370601936

### ADDITIONAL INFORMATION
